packages/core-flows/src/definition/cart/steps/get-variant-price-sets.ts

```diff
@@ -0,0 +1,70 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import { IPricingModuleService } from "@medusajs/types"
+import { MedusaError } from "@medusajs/utils"
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
+
+interface StepInput {
+  variantIds: string[]
+  context: {
+    region_id?: string
+    currency_code?: string
+  }
+}
+
+export const getVariantPriceSetsStepId = "get-variant-price-sets"
+export const getVariantPriceSetsStep = createStep(
+  getVariantPriceSetsStepId,
+  async (data: StepInput, { container }) => {
+    if (!data.variantIds.length) {
+      return new StepResponse({})
+    }
+
+    const pricingModuleService = container.resolve<IPricingModuleService>(
+      ModuleRegistrationName.PRICING
+    )
+
+    const remoteQuery = container.resolve("remoteQuery")
+
+    const variantPriceSets = await remoteQuery(
```

Comment:
@carlos-r-l-rodrigues, could also make sense to have your eyes on this. What do you think about this approach?

Reply:
you can use variables with the json syntax as well.
so we don't have to use gql strings

```
await remoteQuery({
  variant {
    fields: ["id"],
    price {
      fields: ["price_set_id"]
    }
  }
}, {
   variant: {
     id: data.variantIds,
   }
})
```

Reply:
also I would like to have this implemented in the RemoteQuery:

```
await remoteQuery({
  variant {
    fields: ["id"],
    price {
      fields: ["price_set_id"]
    }
  }
}, {
   variant: {
     id: data.variantIds,
   },
   "variant.price": {
     context: data.context
   }
})
```

in case we have a parameter context, instead of calling `list`, it could call a `calculated` method on the module

Reply:
I also think the GQL looks easier to read, but when we get to the point of using it in endpoints, with the option to extend fields, it will be harder to work with strings.
But this is being used internally, so I don't really have a strong opinion.
you'll just have to add the variable on the `query ($var: Any)` on the gql query. 

Regarding `calculate` method, it is up to us to decide where to set this. The same way we call `listAndCount` if there is any pagination parameter we call it. We could call calculate if we have the property `context`.

Reply:
Makes sense. I guess these additional methods on the `remoteQuery` vary from link to link, which is why I thought it would make sense to put them in the link-definition.

Reply:
we can find somewhere in the joiner config to set these method. Because this is more a config of the module itself to calculate something based on parameters, not the link itself.
This case the link variant -> price_set doesn't change. what changes is just the amount of prices that will return depending on the context passed.